content/hoon/lagoon/math.md

```diff
@@ -0,0 +1,1105 @@
+## `+max` {#max}
+
+Yield the maximum value in an array.
+
+#### Accepts
+
+A `$ray`.
+
+#### Produces
+
+See source for return type details.
+
+#### Source
+
+```hoon
+++  max
+  ~/  %max
+  |=  a=ray
+  ?>  (check a)
+  =/  fun
+    |:  [b=1 c=-:(ravel a)]
+    ?.  =(((fun-scalar meta.a %gth) b c) 0)
+      b  c
+  (scalar-to-ray meta.a (reel (ravel a) fun))
+```
+
+---
+
+## `+argmax` {#argmax}
+
+Yield the index of the maximum value in an array.  Only returns the first match.
+
+#### Accepts
+
+A `$ray`.
+
+#### Produces
+
+A `@ud`
+
+#### Source
+
+```hoon
+++  argmax :: Only returns first match
+  ~/  %argmax
+  |=  a=ray
+  ^-  @ud
+  ?>  (check a)
+  +:(find ~[(get-item (max a) ~[0 0])] (ravel a))
+```
+
+---
+
+## `+min` {#min}
+
+Yield the minimum value in an array.
+
+#### Accepts
+
+A `$ray`.
+
+#### Produces
+
+See source for return type details.
+
+#### Source
+
+```hoon
+++  min
+  ~/  %min
+  |=  a=ray
+  ?>  (check a)
+  =/  fun
+    |:  [b=1 c=-:(ravel a)]
+    ?.  =(((fun-scalar meta.a %lth) b c) 0)
+      b  c
+  (scalar-to-ray meta.a (reel (ravel a) fun))
+```
+
+---
+
+## `+argmin` {#argmin}
+
+Yield the index of the minimum value in an array.  Only returns the first match.
+
+#### Accepts
+
+A `$ray`.
+
+#### Produces
+
+A `@ud`
+
+#### Source
+
+```hoon
+++  argmin :: Only returns first match
+  ~/  %argmin
+  |=  a=ray
+  ^-  @ud
+  ?>  (check a)
+  +:(find ~[(get-item (min a) ~[0 0])] (ravel a))
+```
+
+---
+
+## `+cumsum` {#cumsum}
+
+Yield the cumulative sum of an array.
+
+#### Accepts
+
+A `$ray`.
+
+#### Produces
+
+A `$ray`.
+
+#### Source
+
+```hoon
+++  cumsum
+  ~/  %cumsum
+  |=  a=ray
+  ^-  ray
+  ?>  (check a)
+  %+  scalar-to-ray  meta.a
+  (reel (ravel a) |=([b=@ c=@] ((fun-scalar meta.a %add) b c)))
+```
+
+---
+
+## `+prod` {#prod}
+
+Yield the cumulative product of an array.
+
+#### Accepts
+
+A `$ray`.
+
+#### Produces
+
+A `$ray`.
+
+#### Source
+
+```hoon
+++  prod
+  |=  a=ray
+  ^-  ray
+  ?>  (check a)
+  %+  scalar-to-ray  meta.a
+  (reel (ravel a) |=([b=_1 c=_1] ((fun-scalar meta.a %mul) b c)))
+```
+
+---
+
+## `+diag` {#diag}
+
+Return the diagonal of an array.  Enforces that the array is square, and returns a 2D ($n \times 1$) array of the diagonal elements.
+
+#### Accepts
+
+A `$ray`.
+
+#### Produces
+
+A `$ray`.
+
+#### Source
+
+```hoon
+++  diag
+  ~/  %diag
+  |=  a=ray
+  ^-  ray
+  ?>  (check a)
+  =,  meta.a
+  ?>  =(2 (lent shape))
+  ?>  =(-.shape +<.shape)
+  ^-  ray
+  %-  en-ray
+  ^-  baum
+  :-  `meta`[~[-.shape 1] bloq kind tail]
+  ^-  ndray
+  %+  turn
+    `(list @)`(flop (gulf 0 (dec -.shape)))
+  |=(i=@ (get-item a ~[i i]))
+```
+
+---
+
+## `+trace` {#trace}
+
+Compute the trace of an array.
+
+#### Accepts
+
+A `$ray`.
+
+#### Produces
+
+A `$ray`.
+
+#### Source
+
+```hoon
+++  trace
+  ~/  %trace
+  |=  a=ray
+  ^-  ray
+  (cumsum (diag a))
+```
+
+---
+
+## `+dot` {#dot}
+
+Function: dot
```

Comment:
Not clear to me what is meant by this. Same applies to any line beginning with "Function: " followed by one word.